crates/db_views/src/report_combined_view.rs

```diff
@@ -318,13 +329,37 @@ impl ReportCombinedQuery {
       );
     }
 
-    // If its not an admin, get only the ones you mod
-    if !user.local_user.admin {
-      query = query.filter(
-        community_actions::became_moderator
-          .is_not_null()
-          .and(report_combined::community_report_id.is_null()),
-      );
+    // Only show reports for communities where the user is a mod
+    let filter_is_mod = community_actions::became_moderator
+      .is_not_null()
+      .and(report_combined::community_report_id.is_null());
+
+    // Filter reports which are only for admins
+    let filter_to_local_admins = post_report::to_local_admins
+      .or(comment_report::to_local_admins)
+      .or(community_report::to_local_admins);
+
+    if user.local_user.admin {
+      // By default admins only see reports with `to_local_admins == true`, for communities
+      // where they are moderator, or reports which have been unresolved for more than 3 days.
+      // With parameter `show_mod_reports` they can also see newer reports in other communities.
+      let show_mod_reports = self.show_mod_reports.unwrap_or_default();
+      if !show_mod_reports {
+        let three_days = Utc::now() - Days::new(3);
+        query = query.filter(
+          filter_to_local_admins
+            .or(filter_is_mod)
+            .or(post_report::published.lt(three_days))
+            .or(comment_report::published.lt(three_days))
+            .or(community_report::published.lt(three_days)),
+        );
+      }
+    } else {
+      // Mods can only see reports for communities where they are moderator, and
+      // which have `to_local_admins == false`.
+      query = query
+        .filter(filter_is_mod)
+        .filter(filter_to_local_admins.is_distinct_from(true));
```

Comment:
I tried to move this into a helper function `fn filter(&mut query)` but they generic parameters get very confusing. Worst case we have to copy-paste this whole filter logic into `get_report_count()`.